### DISCUSSION TITLE
How can I persist the SpooledTemporaryFile?

### DISCUSSION THREAD

**Discussion by xiaodiannao** [fastapi/fastapi#8106] [Questions] [answered]
Hi everyone, thanks for this amazing project. I'm testing out the features of FastAPI and I'm stuck uploading files at the moment. Now my question is:

How can I persist the SpooledTemporaryFile?

Is it possible to move the contents of SpooledTemporaryFile to a specific file path and persist the file when it's done uploading?

Thanks a lot and best regards

**Comment by dmontagu** [comment, marked as answer]
@xiaodiannao I haven't tested this snippet, but I use code similar to this in one of my projects and you may find it to be a useful starting point:
```python
import shutil

from fastapi import FastAPI, UploadFile, File

app = FastAPI()
@app.post("/upload")
def upload(upload_file: UploadFile = File(...)):
    with open("/path/to/destination.jpg", "wb") as f_destination:
        shutil.copyfileobj(upload_file.file, f_destination)
```

**Comment by xiaodiannao** [comment]
Works like a charm @dmontagu thanks a lot!

**Comment by tiangolo** [comment]
Thanks for the help here @dmontagu ! :clap: :bow: 

Thanks for reporting back and closing the issue @xiaodiannao  :+1:
